src/app/client/src/app/modules/shared/pipes/transposeTerms/transposeTerms.pipe.spec.ts

```diff
@@ -0,0 +1,226 @@
+import { Pipe,PipeTransform } from '@angular/core';
+import { get } from 'lodash-es';
+import { GenericResourceService } from '../../services/genericResource/genericResource.service';
+import { TransposeTermsPipe } from './transposeTerms.pipe';
+
+describe('TransposeTermsPipe', () => {
+    let transposeTermsPipe: TransposeTermsPipe;
+    const genericResourceService :Partial<GenericResourceService> ={};
+
+    beforeAll(() => {
+        transposeTermsPipe = new TransposeTermsPipe(
+            genericResourceService as GenericResourceService
+        )
+    });
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        jest.resetAllMocks();
+    });
+            
+    it('should create a instance of transposeTermsPipe', () => {
+        expect(transposeTermsPipe).toBeTruthy();
+    });
+    
+    describe('transform', () => {
+        it('should return the transformed value when a term is found', () => {
+          const value = 'This is {term1} and {term2}.';
+          const defaultValue = 'Default';
+          const selectedLang = 'en';
+          const transformedValue = transposeTermsPipe.transform(
+            value,
+            defaultValue,
+            selectedLang
+          );
+          expect(transformedValue).toBe('This is {term1} and {term2}.');
+        });
+    
+        it('should return the original value when no terms are found', () => {
+          const value = 'No terms here.';
+          const defaultValue = 'Default';
+          const selectedLang = 'en';
+          const transformedValue = transposeTermsPipe.transform(
+            value,
+            defaultValue,
+            selectedLang
+          );
+          expect(transformedValue).toBe(value);
+        });
+    
+        it('should return the default value when the value is falsy', () => {
+          const value = '';
+          const defaultValue = 'Default';
```

Comment:
can the values be added in beforeAll and not declare in each one it block

Reply:
The repeated values are not used in each testcase with same value so they are added under describe block rather than beforeAll and the code is optimized